### ISSUE TITLE
`Unexpected token` error when folder name contains parentheses

### ISSUE THREAD

**Issue by apptaro** [ezolenko/rollup-plugin-typescript2#94]
#### What happens and why it is wrong

`rollup -c` results in a wrong `Error: Unexpected token` error when the project folder name contains parentheses like `D:\Rollup(Test)`. Workaround is to remove parentheses from the project folder name like `D:\RollupTest`

#### Environment 
- Windows 10
- Node 8.9.4

#### Versions
- typescript: 2.9.2
- rollup: 0.62.0
- rollup-plugin-typescript2: 0.15.1

#### rollup.config.js
```js
import typescript2 from 'rollup-plugin-typescript2';

export default {
    input: 'src/App.ts',
    output: {
        file: 'dist/Apps.js',
        format: 'amd'
    },
    plugins: [
        typescript2({
            clean: true,
            verbosity: 3
        })
    ]
}
```

#### tsconfig.json
```json
{
  "compilerOptions": {
    "target": "es5",
    "module": "es2015"
  }
}
```

#### package.json
```json
{
  "name": "rolluptest",
  "version": "1.0.0",
  "devDependencies": {
    "rollup": "^0.62.0",
    "rollup-plugin-typescript2": "^0.15.1",
    "typescript": "^2.9.2"
  }
}
```

#### plugin output with verbosity 3

<details>
  <summary> log: </summary>

```
D:\Rollup(Test)>node_modules\.bin\rollup -c

src/App.ts → dist/Apps.js...
rpt2: typescript version: 2.9.2
rpt2: tslib version: 1.9.2
rpt2: rollup-plugin-typescript2 version: 0.15.1
rpt2: plugin options:
{
    "clean": true,
    "verbosity": 3,
    "check": true,
    "cacheRoot": "D:\\Rollup(Test)/.rpt2_cache",
    "include": [
        "*.ts+(|x)",
        "**/*.ts+(|x)"
    ],
    "exclude": [
        "*.d.ts",
        "**/*.d.ts"
    ],
    "abortOnError": true,
    "rollupCommonJSResolveHack": false,
    "typescript": "version 2.9.2",
    "useTsconfigDeclarationDir": false,
    "tsconfigOverride": {},
    "transformers": [],
    "tsconfigDefaults": {}
}
rpt2: rollup config:
{
    "external": [
        "",
        ""
    ],
    "inlineDynamicImports": false,
    "input": "src/App.ts",
    "chunkGroupingSize": 5000,
    "perf": false,
    "plugins": [
        {
            "name": "rpt2"
        }
    ],
    "entry": "src/App.ts"
}
rpt2: built-in options overrides: {
    "noEmitHelpers": false,
    "importHelpers": true,
    "noResolve": false,
    "noEmit": false,
    "inlineSourceMap": false,
    "outDir": "D:\\Rollup(Test)/.rpt2_cache/placeholder",
    "moduleResolution": 2,
    "declarationDir": null
}
rpt2: parsed tsconfig: {
    "options": {
        "module": 5,
        "target": 1,
        "configFilePath": "D:/Rollup(Test)/tsconfig.json",
        "noEmitHelpers": false,
        "importHelpers": true,
        "noResolve": false,
        "noEmit": false,
        "inlineSourceMap": false,
        "outDir": "D:\\Rollup(Test)/.rpt2_cache/placeholder",
        "moduleResolution": 2,
        "declarationDir": null
    },
    "fileNames": [
        "D:/Rollup(Test)/src/App.ts"
    ],
    "typeAcquisition": {
        "enable": false,
        "include": [],
        "exclude": []
    },
    "raw": {
        "compilerOptions": {
            "module": "es2015",
            "target": "es5"
        }
    },
    "errors": [],
    "wildcardDirectories": {
        "d:/rollup(test)": 1
    },
    "compileOnSave": false,
    "configFileSpecs": {
        "includeSpecs": [
            "**/*"
        ],
        "validatedIncludeSpecs": [
            "**/*"
        ],
        "wildcardDirectories": {
            "d:/rollup(test)": 1
        }
    }
}
rpt2: included:
'[
    "*.ts+(|x)",
    "**/*.ts+(|x)"
]'
rpt2: excluded:
'[
    "*.d.ts",
    "**/*.d.ts"
]'
rpt2: Ambient types:
rpt2:     D:/Rollup(Test)/node_modules/@types/estree/index.d.ts
rpt2:     D:/Rollup(Test)/node_modules/@types/node/index.d.ts
rpt2: ambient types changed, redoing all semantic diagnostics
[!] Error: Unexpected token
src\App.ts (2:11)
1: export class App {
2:     public text: string = "a";
              ^
3:     constructor() {
4:     }


D:\Rollup(Test)>
```

</details>

**Comment by agilgur5** [comment]
I created a minimal reproduction [here](https://stackblitz.com/edit/rpt2-issue-94) and was unable to reproduce this error.

Neither `tsc` nor rpt2 seem to give this error in their latest versions (`4.7.3` for TS, `0.32.1` for rpt2).

It's possible that this is a Windows-only error, and, unfortunately, I don't have a Windows repro environment to check that. Both this and a downstream issue linked here (https://github.com/PeculiarVentures/webcrypto-liner/issues/71) are on Windows.

The downstream issue is newer and so has some nicer logging, and there the error is coming from Rollup _itself_, not from rpt2. So if this issue _is_ reproducible on Windows, I would try it without rpt2 and just with raw Rollup, and see if it occurs again. If so, then this issue should be filed upstream in Rollup's repo

**Comment by agilgur5** [comment]
Correction, this seems to have been upstream in [`@rollup/pluginutils`](https://github.com/rollup/plugins/tree/master/packages/pluginutils) (not Rollup itself), who's [`createFilter` function](https://github.com/rollup/plugins/tree/master/packages/pluginutils#createfilter) is responsible for filtering files for plugins.

It seems to have had a bug in it per https://github.com/rollup/plugins/issues/80 where it [interpreted parentheses as glob characters](https://github.com/rollup/plugins/issues/80#issuecomment-563015738), and so filtered out the file from any plugin trying to read it. No plugin read it, so Rollup tries to read it, and then gives an error that this file is not JS and that plugins are needed to read it.

This bug was fixed upstream in https://github.com/rollup/plugins/pull/84 (and some similar bugs in https://github.com/rollup/plugins/pull/141) and released in `@rollup/pluginutils` [v3.0.1+](https://github.com/rollup/plugins/blob/master/packages/pluginutils/CHANGELOG.md#301).

rpt2 was updated to a compatible version in https://github.com/ezolenko/rollup-plugin-typescript2/pull/219 (notably, same author that fixed that bug) and released in [`0.27.0`](https://github.com/ezolenko/rollup-plugin-typescript2/releases/0.27.0).

So rpt2 `0.27.0`+ should no longer have this bug.
